dashboard/legacy/middleware/helpers/bucket_helper.rb

```diff
@@ -165,7 +165,9 @@ def restore_file_version(encrypted_channel_id, filename, version)
     owner_id, storage_app_id = storage_decrypt_channel_id(encrypted_channel_id)
     key = s3_path owner_id, storage_app_id, filename
 
-    s3.copy_object(bucket: @bucket, copy_source: ERB::Util.url_encode("#{@bucket}/#{key}?versionId=#{version}"), key: key, metadata_directive: 'REPLACE')
+    encoded_location = ERB::Util.url_encode(@bucket + '/' + key)
+    copy_source = encoded_location + '?versionId=' + version
```

Comment:
tiny nit: prefer string interpolation over concatenation
```suggestion
    copy_source = "#{encoded_location}?versionId=#{version}"
```

Reply:
I find the opposite more readable at times, especially if you're inserting a single character like a slash :) Someone from student learning, break the tie!

Reply:
In the absence of a tiebreaker, tie goes to the reviewer. Updated!

Reply:
Lol, charitable, thanks!

FWIW, I'm not _opposed_ to using concatenation here and I definitely get where you're coming from with it being more readable for simple cases, but I just have a general preference for consistency. Case-by-case styling decisions can make long-term maintenance more confusing.